### PR TITLE
Add page to preferences for general settings

### PR DESCRIPTION
**Pull request by FroggieFrog**
This PR adds a page for general settings to the preferences window.
This PR is not responsible for adding the actual functionality. Those will be added with different PRs.

##### What is missing:

- [x] localizaion
- [x] update localization on language change (comboboxes)
- [x] functionality of all settings
  - [x] add zoom to point (PR #220)
  - [x] light gray grid lines (issue #211 | PR #)
  - [x] add option to hide influence on selction (issue #205 | this PR)

grid_lines_color

### CODE REVIEW

File: AnnoDesigner/Models/UserDefinedColorType.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AnnoDesigner.Models
+{
+    public enum UserDefinedColorType
```
Comment (AgmasGold) [inline AnnoDesigner/Models/UserDefinedColorType.cs:9]: Might be worth adding a comment here that a string value of `UserDefinedColorType` is used for localization.

File: AnnoDesigner/Models/UserDefinedColor.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+namespace AnnoDesigner.Models
+{
+    [DebuggerDisplay("{" + nameof(Type) + ",nq}")]
+    [DataContract]
+    public class UserDefinedColor : Notify
+    {
+        private UserDefinedColorType _type;
+        private SerializableColor _color;
+
+        [DataMember(Order = 0)]
```
Comment (AgmasGold) [inline AnnoDesigner/Models/UserDefinedColor.cs:14]: Do we need these `DataMember` and `DataContract` attributes? Especially as this is in settings anyway, so won't be normally seen by the user. I just feel like they add extra stuff that new developers may think are important when they are actually not.

Comment (FroggieFrog) [reply AnnoDesigner/Models/UserDefinedColor.cs:14]: I explicitly added them to make it more obvious which properties would be serialized.
But I also can remove them.

**Comment by FroggieFrog** [comment]
> 
> 
> `UpdateBorderLineColorVisibility` and `UpdateGridLineColorVisibility` seem to use the same pattern of logic - is there a way to simplify the logic into a single method?

I  created separate methods, because there could be the case where the entries of the comboboxes are different.
So one has only 2 entries and the other has all entries or so.

File: AnnoDesigner/ViewModels/GeneralSettingsViewModel.cs [hunk 3]
```diff
@@ -45,6 +48,19 @@ public GeneralSettingsViewModel(IAppSettings appSettingsToUse)
             SelectedCustomObjectBorderLineColor = savedObjectBorderLineColor.Color;
         }
 
+        private void Commons_SelectedLanguageChanged(object sender, EventArgs e)
+        {
+            var selectedGridLineColorType = SelectedGridLineColor.Type;
+            GridLineColors.Clear();
+            InitGridLineColors();
```
Comment (AgmasGold) [inline AnnoDesigner/ViewModels/GeneralSettingsViewModel.cs:55]: Maybe these methods should be renamed to `RefreshGridLIneColors` and `RefreshObjectBorderLineColors` seeing as they are no longer being called only once to initialise the combo boxes when the window is constructed?

File: AnnoDesigner/Models/UserDefinedColor.cs (hunk 2 above)
Comment (AgmasGold) [reply AnnoDesigner/Models/UserDefinedColor.cs:14]: Yeah, I would just remove them.

**Pull request merged**